bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/service/impl/BahmniPatientServiceImpl.java

```diff
@@ -93,4 +99,30 @@ public PatientConfigResponse getConfig() {
         return patientDao.getByAIsToB(aIsToB);
     }
 
+    @Override
+    public List<PatientResponse> luceneHibernateSearch(PatientSearchParameters searchParameters) {
```

Comment:
Comment: I find that the method naming here makes this very hard to understand. How would anyone possibly know what luceneSearch, lucenseHibernateSearch, and search are suppose to do, and which ones handle which parameters?
Can we deprecate and/or remove some methods? E.g. if the point of this ticket is to make a faster-but-otherwise-equivalent version of "search" then let's use that name for the new behavior. If some of that old code needs to go into a private method, that's fine.